routing/route_weight.cpp

```diff
@@ -41,11 +41,11 @@ RouteWeight RouteWeight::operator+(RouteWeight const & rhs) const
 
 RouteWeight RouteWeight::operator-(RouteWeight const & rhs) const
 {
-  ASSERT_NOT_EQUAL(m_numPassThroughChanges, std::numeric_limits<int32_t>::min(), ());
-  ASSERT_NOT_EQUAL(m_numAccessChanges, std::numeric_limits<int32_t>::min(), ());
-  ASSERT(!SumWillOverflow(m_numPassThroughChanges, -rhs.m_numPassThroughChanges),
+  ASSERT_NOT_EQUAL(m_numPassThroughChanges, numeric_limits<int8_t>::min(), ());
+  ASSERT_NOT_EQUAL(m_numAccessChanges, numeric_limits<int8_t>::min(), ());
+  ASSERT(!SumWillOverflow(m_numPassThroughChanges, static_cast<int8_t>(-rhs.m_numPassThroughChanges)),
```

Comment:
I guess it's safer to cast first and negate second. I.e.:
```c++
-static_cast<int8_t>(...)
```

Reply:
BTW, numPassThroughChanges is int8_t, do we really need cast?

Reply:
Hello @ygorshenin :)

rhs.m_numPassThroughChanges has type int8_t. -rhs.m_numPassThroughChanges gets a type differ form int8_t because the template method
```
template <typename Number, typename EnableIf = enable_if_t<is_integral<Number>::value, void>>
bool SumWillOverflow(Number lhs, Number rhs)
```
was not instantiated. So I guess`-static_cast<int8_t>(...)` gets a type differ from int8_t.

Reply:
Yes, seems that int8_t will be promoted.

Reply:
Hello, @bykoianko !